src/Dialect/Mlir/VectorMachineSupport.hpp

```diff
@@ -63,7 +64,11 @@ enum class GenericOps {
   TrigHyperbolicGop, /* Hyperbolic trig. */
 };
 
+// Describe the mix of Generic operations in a given kernel. Each generic
+// operation is associated with a number, which indicates the number of
+// occurrence of that generic op in the given kernel.
 using GenOpsMix = mlir::ArrayRef<std::pair<GenericOps, int64_t>>;
+using GenOpsMixList = llvm::SmallVector<std::pair<GenericOps, int64_t>, 4>;
```

Comment:
Can std::map be used for GenOpsMix?

Reply:
Another good point, as long as I can iterate through all the keys, it would work too and be more elegant.

Reply:
Good suggestion, needed a bit of code change but I like it. Done.